mm2src/coins/lightning/ln_platform.rs

```diff
@@ -0,0 +1,688 @@
+            )
+            .compat()
+            .await
+            .map_to_mm(SaveChannelClosingError::WaitForFundingTxSpendError)?;
+
+        let closing_tx_hash = format!("{:02x}", closing_tx.tx_hash());
+
+        Ok(closing_tx_hash)
+    }
+}
+
+impl FeeEstimator for Platform {
+    // Gets estimated satoshis of fee required per 1000 Weight-Units.
+    fn get_est_sat_per_1000_weight(&self, confirmation_target: ConfirmationTarget) -> u32 {
+        let platform_coin = &self.coin;
+
+        let default_fee = match confirmation_target {
+            ConfirmationTarget::Background => self.default_fees_and_confirmations.background.default_feerate,
+            ConfirmationTarget::Normal => self.default_fees_and_confirmations.normal.default_feerate,
+            ConfirmationTarget::HighPriority => self.default_fees_and_confirmations.high_priority.default_feerate,
+        } * 4;
+
+        let conf = &platform_coin.as_ref().conf;
+        let n_blocks = match confirmation_target {
+            ConfirmationTarget::Background => self.default_fees_and_confirmations.background.n_blocks,
+            ConfirmationTarget::Normal => self.default_fees_and_confirmations.normal.n_blocks,
+            ConfirmationTarget::HighPriority => self.default_fees_and_confirmations.high_priority.n_blocks,
+        };
+        let fee_per_kb = tokio::task::block_in_place(move || {
+            platform_coin
+                .as_ref()
+                .rpc_client
+                .estimate_fee_sat(
+                    platform_coin.decimals(),
+                    // Todo: when implementing Native client detect_fee_method should be used for Native and
+                    // EstimateFeeMethod::Standard for Electrum
+                    &EstimateFeeMethod::Standard,
+                    &conf.estimate_fee_mode,
+                    n_blocks,
+                )
+                .wait()
+                .unwrap_or(default_fee)
+        });
+        // Must be no smaller than 253 (ie 1 satoshi-per-byte rounded up to ensure later round-downs don’t put us below 1 satoshi-per-byte).
+        // https://docs.rs/lightning/0.0.101/lightning/chain/chaininterface/trait.FeeEstimator.html#tymethod.get_est_sat_per_1000_weight
+        cmp::max((fee_per_kb as f64 / 4.0).ceil() as u32, MIN_ALLOWED_FEE_PER_1000_WEIGHT)
+    }
+}
+
+impl BroadcasterInterface for Platform {
+    fn broadcast_transaction(&self, tx: &Transaction) {
+        let txid = tx.txid();
+        let fut = Box::new(async move { self.check_if_tx_is_onchain(txid).await }.boxed().compat());
+        let is_tx_onchain = tokio::task::block_in_place(move || fut.wait());
+
+        if let Ok(true) = is_tx_onchain {
+            log::debug!("Transaction {} is found onchain!", txid);
+            return;
+        }
+
+        let tx_hex = serialize_hex(tx);
+        log::debug!("Trying to broadcast transaction: {}", tx_hex);
+        let fut = self.coin.send_raw_tx(&tx_hex);
+        spawn(async move {
```

Comment:
Do I understand correctly that we can move the whole method implementation into `spawn(async move {})`? I think so because the transaction is actually sent on the first poll in the spawned future.
So we can avoid using `block_in_place` :)

Reply:
Actually, I will remove `check_if_tx_is_onchain` from here, to reduce the number of requests. Its only purpose was to remove some of the error logs if the transaction was already broadcasted, but there is no problem in trying to broadcast an already broadcasted transaction, plus I still get error logs for input already spent sometimes so it didn't remove all the false-positive errors.